core/d2l-content-uploader/src/util/media-type-util.js

```
@@ -2,7 +2,9 @@ const SUPPORTED_AUDIO_EXTENSIONS = ['.m4a', '.mp3', '.ogg', '.wav', '.wma'];
 const SUPPORTED_VIDEO_EXTENSIONS = ['.avi', '.f4v', '.flv', '.m4v', '.mov', '.mp4', '.webm', '.wmv'];
 
 export function getExtension(filePath) {
-	return filePath.split('.').pop().toLowerCase();
+	const pathParts = filePath.split('.');
+	// ensure that does not mistake an extensionless name for a type
+	return pathParts.length > 1 ? pathParts.pop().toLowerCase() : 'missingExtension';
```

Could we throw (and catch) an error instead of returning 'missingExtension'?

---

Or maybe better - just return `''` and handle that case where this is called?

---

Can it return just `null` - no extension?